Improve event edit modal and fix bugs

0-ft opened the pull request:
This PR makes some UI changes to the event editing modal to make it cleaner and easier to understand:  

## Old:
![image](https://user-images.githubusercontent.com/98317611/169091987-12b9de11-4119-4371-8c6d-4b0ff8b13a50.png)

## New:
![image](https://user-images.githubusercontent.com/98317611/169091922-1abf250c-19c9-4a90-86c3-d592b4e1278e.png)

- Inputs are styled to be more in-line with obsidian (with the exception of the checkbox which ideally would be made into a `ToggleComponent`).
- The recurring event checkbox is removed and whether the event recurs is determined by whether any repeating days are selected
- Buttons are moved
- Modal title added

It also fixes two bugs:
- the modal does not show the end date for a multi-day event when opened
- created filenames for recurring events have an extra `)` appended

davish reviewed (changes requested):
Thanks so much for submitting this PR! I really appreciate it. I had some smaller questions, but one big thing is that I think the "recurring event" toggle is important. 

Most events are one-off events, and even users who have many recurring events will not be dealing with them very frequently in the modal because once they are set you don't have to make more of them, by definition. It doesn't make sense for the repeating event radios to be visible. In fact, it might end up causing a decent amount of accidental events if users interact with the repeating event radios instead of the datepicker, assuming that clicking Monday will make an event occur on "this monday".

Hope that makes sense! Thanks again, and let me know if you have any questions.

0-ft commented:
> one big thing is that I think the "recurring event" toggle is important.

I agree with what you said - I looked at a couple of different event modals from different apps and they vary in their approach, but happy to go with the one you suggest. I've changed this back to include the toggle, and have the week day selector appear only when it's checked (as before).

Other comments also implemented.

davish commented on src/EditModal.css:
```diff
@@ -0,0 +1,51 @@
+    flex-direction: column;
+}
+
+.ofc-edit-modal-event-save-button {
```
I wonder if the save button could be a primary interactive button here, to make it stand out from the delete button and the open note button?

davish reviewed (approved):
Looks great! One comment about making the safe button more prominent that could be good to do before merging. Im staying afk from my laptop for the weekend, but will try this out locally when I can and merge if it looks good.
